src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageService.cs

```diff
@@ -222,15 +224,21 @@ private MemoryMappedInfo CreateTemporaryStorage(long size)
             else
             {
                 // Reserve additional space in the existing storage location
-                Contract.ThrowIfNull(_name);
                 _offset += size;
                 return new MemoryMappedInfo(reference, _name, _offset - size, size);
             }
         }
     }
 
-    public static string CreateUniqueName(long size)
-        => "Roslyn Temp Storage " + size.ToString() + " " + Guid.NewGuid().ToString("N");
+    public static string? CreateUniqueName(long size)
+    {
+        // MemoryMapped files which are used by the TemporaryStorageService are present in .NET Framework (including Mono)
+        // and .NET Core Windows. For non-Windows .NET Core scenarios, we can return the TrivialTemporaryStorageService
```

Comment:
comment probably needs to be updated since the trivial service is deleted.

Reply:
yup.  will do.